gatsby-node.js

```diff
@@ -75,26 +77,18 @@ exports.sourceNodes = async () => {
         console.error('No documents matched your query.');
     }
 
-    documents.forEach(doc => {
-        const { page_id, ...rest } = doc;
-        RESOLVED_REF_DOC_MAPPING[
-            page_id.replace(`${PAGE_ID_PREFIX}/`, '')
-        ] = rest;
-    });
-
-    // Identify page documents and parse each document for images
     const assets = [];
-    Object.entries(RESOLVED_REF_DOC_MAPPING).forEach(([key, val]) => {
-        const pageNode = getNestedValue(['ast', 'children'], val);
-        const filename = getNestedValue(['filename'], val) || '';
-        if (pageNode) {
-            assets.push(...val.static_assets);
-        }
-        if (key.includes('images/')) {
-            IMAGE_FILES[key] = val;
-        } else if (filename.endsWith('.txt')) {
-            PAGES.push(key);
-        }
+    documents.forEach(doc => {
+        // Mimics onCreateNode
+        const { page_id, ...nodeContent } = doc;
+        const slug = page_id.replace(`${PAGE_ID_PREFIX}/`, '');
+        postprocessDocument(
```

Comment:
I chose to keep the in-memory array updates here to help this function mimic a potential [`onCreateNode` callback](https://www.gatsbyjs.org/docs/node-apis/#onCreateNode), should we move to using `createNode` to populate a data layer. This would be an easy switch.

Reply:
What do you think about passing the `doc` to `postprocessDocument` and getting the `slug` and `nodeContent` in that function? Is that something that would clash with your goal of mimicing `onCreateNode`?

Reply:
Or is this actually the cleaner approach? Hard to say with the different params passing around.

Reply:
Yeah we can pull in the logic to get the slug. I was concerned about continuing to pass the `PAGE_ID_PREFIX`, but it would be nice to bring all the logic into one place.